pytorch_lightning/trainer/trainer.py

```diff
@@ -1038,12 +1041,11 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
         self._pre_dispatch()
 
         if self.accelerator.restore_checkpoint_after_pre_dispatch:
-            if self._ckpt_path:
-                self._load_checkpoint_weights()
-            self._restore_modules_and_callbacks()
+            self._restore_modules_and_callbacks(ckpt_path)
 
         # restore optimizers, etc.
-        self.checkpoint_connector.restore_training_state()
+        if self.state.fn == TrainerFn.FITTING:
+            self.checkpoint_connector.restore_training_state()
```

Comment:
restore training state includes things which can be resumed even if not fitting, such as the loop state.

imo we shouldn't add the check for fitting here, but rather inside the select parts inside of `restore_training_state`

Reply:
(above comment)

Reply:
yeah, we do restore loops there and now some other attributes as well. I'd suggest waiting for this one to get merged: https://github.com/PyTorchLightning/pytorch-lightning/pull/9413